src/components/register-in-contest.jsx

```diff
@@ -79,79 +105,64 @@ export default React.createClass({
+    return <div>
       Termină ceilalți pași înainte să îl completezi pe acesta.
     </div>;
   },
 
   renderContestantForm() {
-    if (this.props.current.registration.has_contestant) {
-      return this.renderSuccess();
-    }
-    else {
-      return <RegisterContestant current={this.props.current}
-                                 hasSubmited={this.submitProject} />;
-    }
+    return <RegisterContestant access_token={this.props.user.access_token}
+                               onSubmit={this.props.refreshCurrent} />;
   },
 
-  renderRegisterProjectForm() {
-    if (this.props.current.registration.has_pending_project ||
-        this.props.current.registration.has_projects) {
-      return this.renderSuccess();
-    }
-    else if (!this.props.current.registration.has_contestant) {
-      return this.renderUnavailableStep();
-    }
-    else {
-      return <RegisterProject current={this.props.current}
-                              hasSubmited={this.submitProject} />;
-    }
+  renderWIPStep() {
+    return <p>
+      Încă nu e gata...
+    </p>;
+  },
+
+  renderProjectForm() {
+    return <RegisterProject access_token={this.props.user.access_token}
+                            onSubmit={this.props.refreshCurrent} />;
   },
 
   renderFinishForm() {
-    let formEndpoint = "";
-    if (this.props.current.registration.has_contestant) {
-      formEndpoint = `projects/${this.props.current.registration.pending_project.id}/finish`;
-    }
-    if (this.props.current.registration.has_pending_project) {
-      return <RegisterFinish current={this.props.current}
-                             hasSubmited={this.submitFinish}
-                             formEndpoint={formEndpoint} />;
-    }
-    if (!this.props.current.registration.has_pending_project &&
-        this.props.current.registration.has_projects) {
-      return this.renderSuccess();
-    }
-    if (!this.props.current.registration.has_pending_project &&
-        !this.props.current.registration.has_projects) {
-      return this.renderUnavailableStep();
-    }
+    let formEndpoint = `projects/${this.props.registration.pending_project.id}/finish`;
+    return <RegisterFinish access_token={this.props.user.access_token}
+                           pending_project_title={this.props.registration.pending_project.title}
+                           onSubmit={this.props.refreshCurrent}
+                           formEndpoint={formEndpoint} />;
   },
 
   renderRegisteredProjects() {
-    if (!this.props.current.registration.has_contestant ||
-        !this.props.current.registration.projects.length) {
+    if (!this.props.registration.finished_projects.length) {
       return <p>Nu ai niciun proiect înscris.</p>;
     }
-    return <div><p>Proiectele inscrise de tine până acum:</p><ul>
-      {this.props.current.registration.projects.map((project) => {
-        if (project.finished) {
+    return <div>
+      <p>Proiectele inscrise de tine până acum:</p>
```

Comment:
Eventual sa le si zicem ca sunt in asteptare sa fie aprobate.
`<p>Proiectele inscrise de tine până acum care așteaptă să fie aprobate.</p>`